tools/bin/build_image.sh

```diff
@@ -42,7 +42,7 @@ if [ "$FOLLOW_SYMLINKS" == "true" ]; then
   # to use as the build context
   tar cL "${exclusions[@]}" . | docker build - "${args[@]}"
 else
-  JDK_VERSION="${JDK_VERSION:-17.0.1}"
+  JDK_VERSION="${JDK_VERSION:-17}"
```

Comment:
I have no idea if we care about the patch version in practice, but I'd like to stick with what we have already unless we have a good reason to move.  I also see we are using `17.0.4` elsewhere.  

I don't understand fully how these are related to each other in practice though